### PR TITLE
Terraform Database on AWS RDS

### LINKED ISSUE #25: Terraform Database
### Steps
- [x] Write Terraform scripts to deploy the database
- [x] Include necessary configurations and parameters in the scripts
- [x] Test the Terraform scripts for correct deployment
- [x] Document the Terraform deployment process

### User Story
As a cloud engineer, I need Terraform scripts to automate the deployment of the database to ensure consistency and repeatability.

### PR DESCRIPTION
**Pull request by A-Faris**
Terraformed the database on AWS RDS and created output file which shows the information needed to connect to the database. Also updated the README file to explain the terraform.
closes #25

### CODE REVIEW

File: Database/README.md
Comment (BerkayDur) [inline Database/README.md:1]: Looks good. It is very informative while still being easy to understand.

File: Database/Terraform/main.tf [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    instance_class                        = "db.t3.micro"
+    license_model                         = "postgresql-license"
+    network_type                          = "IPV4"
+    port                                  = 5432
+    password                              = var.DB_PASSWORD
+    publicly_accessible                   = true
+    skip_final_snapshot                   = true
+    storage_encrypted                     = true
+    username                              = "postgres"
+    vpc_security_group_ids                = ["sg-0599fe57129262ed0"]
```
Comment (BerkayDur) [inline Database/Terraform/main.tf]: Shouldn't this security group be created in the terraform?

File: Database/Terraform/main.tf [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  region = var.AWS_REGION
+}
+
+# terraform import aws_db_instance.database c11-hermes-db
+resource "aws_db_instance" "database" {
+    allocated_storage                     = 20
+    db_name                               = "sales"
+    availability_zone                     = "eu-west-2a"
+    copy_tags_to_snapshot                 = true
+    db_subnet_group_name                  = "c11-public-subnet-group"
```
Comment (BerkayDur) [inline Database/Terraform/main.tf]: Could this be put into `tfvars?`

File: Database/Terraform/main.tf [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    engine                                = "postgres"
+    engine_version                        = "16.3"
+    identifier                            = "c11-hermes-db"
```
Comment (BerkayDur) [inline Database/Terraform/main.tf]: Could this be put into `tfvars`?

File: Database/Terraform/main.tf [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+provider "aws" {
+  region = var.AWS_REGION
+}
+
+# terraform import aws_db_instance.database c11-hermes-db
+resource "aws_db_instance" "database" {
+    allocated_storage                     = 20
+    db_name                               = "sales"
```
Comment (BerkayDur) [inline Database/Terraform/main.tf]: I think a clearer database name would be nice

Comment (BerkayDur) [reply Database/Terraform/main.tf]: I think also putting this into `tfvars` would help

File: Database/Terraform/main.tf [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    identifier                            = "c11-hermes-db"
+    instance_class                        = "db.t3.micro"
+    license_model                         = "postgresql-license"
+    network_type                          = "IPV4"
+    port                                  = 5432
+    password                              = var.DB_PASSWORD
+    publicly_accessible                   = true
+    skip_final_snapshot                   = true
+    storage_encrypted                     = true
+    username                              = "postgres"
```
Comment (BerkayDur) [inline Database/Terraform/main.tf]: Put this in `tfvars` as it is something that shouldn't be viewable by everyone

File: Database/Terraform/outputs.tf [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# This file outputs data from the database
+# The output data is useful for connecting to the database
+
+data "aws_db_instance" "database" {
+  db_instance_identifier = "c11-hermes-db"
+}
```
Comment (BerkayDur) [inline Database/Terraform/outputs.tf]: Is this part necessary? Could be instead directly access resources from `main.tf`?

**Review by BerkayDur — CHANGES_REQUESTED** [review]

File: Database/Terraform/outputs.tf (hunk 6 above)
Comment (A-Faris) [reply Database/Terraform/outputs.tf]: True this part wasn't needed, thanks

File: Database/Terraform/main.tf (hunk 1 above)
Comment (A-Faris) [reply Database/Terraform/main.tf]: I'll terraform the security group

File: Database/Terraform/main.tf (hunk 2 above)
Comment (A-Faris) [reply Database/Terraform/main.tf]: done

File: Database/Terraform/main.tf (hunk 3 above)
Comment (A-Faris) [reply Database/Terraform/main.tf]: done

File: Database/Terraform/main.tf (hunk 5 above)
Comment (A-Faris) [reply Database/Terraform/main.tf]: done

File: Database/Terraform/main.tf (hunk 4 above)
Comment (A-Faris) [reply Database/Terraform/main.tf]: name changed and done

File: Database/Terraform/main.tf (hunk 1 above)
Comment (A-Faris) [reply Database/Terraform/main.tf]: done

File: Database/Terraform/outputs.tf
Comment (BerkayDur) [inline Database/Terraform/outputs.tf:1]: Looks very good now without the data value.

File: Database/Terraform/main.tf [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+provider "aws" {
+  region = var.AWS_REGION
+}
+
+# terraform import aws_db_instance.database c11-hermes-db
+resource "aws_db_instance" "database" {
+    allocated_storage                     = 20
+    db_name                               = var.DB_NAME
+    availability_zone                     = "eu-west-2a"
```
Comment (BerkayDur) [inline Database/Terraform/main.tf]: I would also put this in `tfvars` as it depends on `var.AWS_REGION`.

File: Database/Terraform/variables.tf [hunk 8]
```diff
@@ -0,0 +1,40 @@
+# This file assigns variables to the values stored in terraform.tfvars
+
+variable "AWS_REGION" {
+  type = string
+  default = "eu-west-2"
```
Comment (BerkayDur) [inline Database/Terraform/variables.tf:5]: Good use of default values.

**Review by BerkayDur — CHANGES_REQUESTED** [review]

File: Database/Terraform/main.tf (hunk 7 above)
Comment (A-Faris) [reply Database/Terraform/main.tf]: done

**Review by BerkayDur — APPROVED** [review]
Seems good.

**Pull request merged**